### PR TITLE
[Add-ins] (attachments) Include preview APIs

### PR DESCRIPTION
**Pull request by ElizabethSamuel-MSFT**

### CODE REVIEW

**Comment by jasonjoh** [comment]
Docs Build status updates of commit _[71686eb](https://github.com/OfficeDev/outlook-dev-docs/commits/71686eb45f382c26cd91822b749f937e1fa16d50)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/add-ins/add-and-remove-attachments-to-an-item-in-a-compose-form.md](https://github.com/OfficeDev/outlook-dev-docs/blob/elizs-attachments/docs/add-ins/add-and-remove-attachments-to-an-item-in-a-compose-form.md) | :white_check_mark:Succeeded | |
[docs/add-ins/delegate-access.md](https://github.com/OfficeDev/outlook-dev-docs/blob/elizs-attachments/docs/add-ins/delegate-access.md) | :white_check_mark:Succeeded | |
[docs/add-ins/get-or-set-the-location-of-an-appointment.md](https://github.com/OfficeDev/outlook-dev-docs/blob/elizs-attachments/docs/add-ins/get-or-set-the-location-of-an-appointment.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C8%5C29%5Cf49385f6-041e-c905-242e-e0025c631156%5CPullRequest%5C201908292135016848-685%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=11jdFfRrOAehSDCJFQekC5aX9Wzv3eS9o4ibDYp3F%2BM%3D&st=2019-08-29T21%3A31%3A02Z&se=2019-09-29T21%3A36%3A02Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by jasonjoh** [comment]
Docs Build status updates of commit _[db40a69](https://github.com/OfficeDev/outlook-dev-docs/commits/db40a6973880865a5e13e9b93c300d11de85b808)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/add-ins/add-and-remove-attachments-to-an-item-in-a-compose-form.md](https://github.com/OfficeDev/outlook-dev-docs/blob/elizs-attachments/docs/add-ins/add-and-remove-attachments-to-an-item-in-a-compose-form.md) | :white_check_mark:Succeeded | |
[docs/add-ins/delegate-access.md](https://github.com/OfficeDev/outlook-dev-docs/blob/elizs-attachments/docs/add-ins/delegate-access.md) | :white_check_mark:Succeeded | |
[docs/add-ins/get-or-set-the-location-of-an-appointment.md](https://github.com/OfficeDev/outlook-dev-docs/blob/elizs-attachments/docs/add-ins/get-or-set-the-location-of-an-appointment.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C8%5C30%5Cf49385f6-041e-c905-242e-e0025c631156%5CPullRequest%5C201908301719444699-685%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=fvX1PM9FIFZLCl3wzhgBBRqckl2f4xTWLWMHvXzLf00%3D&st=2019-08-30T17%3A15%3A47Z&se=2019-09-30T17%3A20%3A47Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/add-ins/add-and-remove-attachments-to-an-item-in-a-compose-form.md [hunk 1]
```diff
@@ -101,33 +111,76 @@ function addItemAttachment(itemId) {
...
+  if (result.value.length > 0) {
+    for (i = 0 ; i < result.value.length ; i++) {
+      result.asyncContext.currentItem.getAttachmentContentAsync(result.value[i].id, handleAttachmentsCallback);
+    }
+  }
+}
+
+function handleAttachmentsCallback(result) {
+  // Parse string to be a url, an .eml file, a base64-encoded string, or an .icalendar file.
+  if (result.value.format === Office.MailboxEnums.AttachmentContentFormat.Base64) {
```
Comment (Rick-Kirkham) [inline docs/add-ins/add-and-remove-attachments-to-an-item-in-a-compose-form.md]: Not a big deal, but consider making this a switch statement. That's usually easier to read than a series of `else if`s.

File: docs/add-ins/add-and-remove-attachments-to-an-item-in-a-compose-form.md [hunk 2]
```diff
@@ -101,33 +111,76 @@ function addItemAttachment(itemId) {
...
+    // Handle cloud attachment.
+  } else {
+    // Handle attachment formats that are not supported.
+  }
+}
+```
+
+## Remove an attachment
+
+You can remove a file or item attachment from a message or appointment item in a compose form by specifying the corresponding attachment ID and using the [removeAttachmentAsync](/office/dev/add-ins/reference/objectmodel/requirement-set-1.5/Office.context.mailbox.item#removeattachmentasyncattachmentid-options-callback) method. You should only remove attachments that the same add-in has added in the same session. You should make sure the attachment ID corresponds to a valid attachment, or the method will return an error. Similar to the `addFileAttachmentAsync` and `addItemAttachmentAsync` methods, `removeAttachmentAsync` is an asynchronous method. You should provide a callback method to check for the status and any error by using the `AsyncResult` output parameter object. You can also pass any additional parameters to the callback method by using the optional `asyncContext` parameter.
```
Comment (Rick-Kirkham) [inline docs/add-ins/add-and-remove-attachments-to-an-item-in-a-compose-form.md]: Do we really  need to say "You should make sure the attachment ID corresponds to a valid attachment, or the method will return an error."?

**Comment by jasonjoh** [comment]
Docs Build status updates of commit _[dc9fc11](https://github.com/OfficeDev/outlook-dev-docs/commits/dc9fc1124dd163c83357f4c0109642be4602b7e7)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/add-ins/add-and-remove-attachments-to-an-item-in-a-compose-form.md](https://github.com/OfficeDev/outlook-dev-docs/blob/elizs-attachments/docs/add-ins/add-and-remove-attachments-to-an-item-in-a-compose-form.md) | :white_check_mark:Succeeded | |
[docs/add-ins/delegate-access.md](https://github.com/OfficeDev/outlook-dev-docs/blob/elizs-attachments/docs/add-ins/delegate-access.md) | :white_check_mark:Succeeded | |
[docs/add-ins/get-or-set-the-location-of-an-appointment.md](https://github.com/OfficeDev/outlook-dev-docs/blob/elizs-attachments/docs/add-ins/get-or-set-the-location-of-an-appointment.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C8%5C30%5Cf49385f6-041e-c905-242e-e0025c631156%5CPullRequest%5C201908302048427543-685%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=DsX33E5X%2BK%2Bu%2Fj%2FHVhaxgBC9btP8rvPxkV3raRzVOqk%3D&st=2019-08-30T20%3A44%3A41Z&se=2019-09-30T20%3A49%3A41Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by jasonjoh** [comment]
Docs Build status updates of commit _[098b51b](https://github.com/OfficeDev/outlook-dev-docs/commits/098b51bbfdb0da6c30fb03a7b1f3a95dec0bf1e5)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/add-ins/add-and-remove-attachments-to-an-item-in-a-compose-form.md](https://github.com/OfficeDev/outlook-dev-docs/blob/elizs-attachments/docs/add-ins/add-and-remove-attachments-to-an-item-in-a-compose-form.md) | :white_check_mark:Succeeded | |
[docs/add-ins/delegate-access.md](https://github.com/OfficeDev/outlook-dev-docs/blob/elizs-attachments/docs/add-ins/delegate-access.md) | :white_check_mark:Succeeded | |
[docs/add-ins/get-or-set-the-location-of-an-appointment.md](https://github.com/OfficeDev/outlook-dev-docs/blob/elizs-attachments/docs/add-ins/get-or-set-the-location-of-an-appointment.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C8%5C30%5Cf49385f6-041e-c905-242e-e0025c631156%5CPullRequest%5C201908302050315082-685%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=p%2Fqic45PJ9xvqTzR9%2BMEtKrGSisvgC0LbZyCg38GBm0%3D&st=2019-08-30T20%3A46%3A32Z&se=2019-09-30T20%3A51%3A32Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by ElizabethSamuel-MSFT** [comment]
@Rick-Kirkham Thanks, I've incorporated your feedback.

**Pull request merged**